File: algorithms/Strings/Easy/ValidAnagram/VSDValidAnagram.cpp
#include<bits/stdc++.h>

using namespace std;

bool isAnagram(string s,string t){

    if(s.length()!=t.length()) return false;
    map<char,int> charCount;
    for(char ch:s){
        if(charCount.find(ch)==charCount.end()) charCount.insert({ch,1});
        else charCount[ch]++;
    }
   
    for( char ch:t){
        if(charCount.find(ch)==charCount.end())return false;
        charCount[ch]--;
    }
    
    for(auto& e:charCount){
        if(e.second!=0) return false;
        return true;
    }
}

int main(){
    if(isAnagram("hello","olleh")) cout<<"These are Anagrams";
    else cout<<"These are not Anagrams";
    return 0;
}

File: algorithms/Strings/Easy/ValidAnagram/VSDValidAnagram.java
import java.util.HashMap;

public class Main {

    static boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
            return false;
        HashMap<Character, Integer> charCount = new HashMap<>();

        for (char ch : s.toCharArray())
            charCount.put(ch, 1 + charCount.getOrDefault(ch, 0));
        for (char ch : t.toCharArray()) {
            if (!charCount.containsKey(ch))
                return false;
            charCount.put(ch, charCount.get(ch) - 1);
        }
        for (var d : charCount.entrySet()) {
            if (d.getValue() != 0)
                return false;

        }
        return true;

    }

    public static void main(String[] args) {
        String s = "hello", t = "olleh";
        if (isAnagram(s, t))
            System.out.println("These are anagrams");
        else
            System.out.println("These are not anagrams");
    }
}